src/server/api/routers/user.ts

```diff
@@ -47,6 +47,32 @@ export const userRouter = createTRPCRouter({
       });
     }),
 
+  //Update the user's preferences in their profile, and is used for the matching page
+  updatePreferences: protectedProcedure
+    .input(
+      z.object({
+        preferences: z.array(z.string()),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      //Update the user's preferences in their profile
+      try {
+        return await db.profile.update({
```

Comment:
We should probably check to make sure each id is valid i.e. is a mentor/mentee and exists. Or else Patrick Hill will come and find you in your sleep.

And so we know that the entries are for sure valid (although it will probably be prudent to check in the algorithm part as well incase of deleted users)